src/tracks/ui/BrushHandle.h

```diff
@@ -0,0 +1,122 @@
+/**********************************************************************
+
+Audacity: A Digital Audio Editor
+
+BrushHandle.h
+
+Paul Licameli split from TrackPanel.cpp
+
+**********************************************************************/
+
+#ifndef __AUDACITY_BRUSH_HANDLE__
+#define __AUDACITY_BRUSH_HANDLE__
+
+#include "../../UIHandle.h"
+#include "../../SelectedRegion.h"
+#include "../../Snap.h"
+
+#include <vector>
+#include <tracks/playabletrack/wavetrack/ui/SpectrumView.h>
```

Comment:
I find a dependency cycle of SpectrumView and BrushHandle, but no other files in that cycle.

I think BrushHandle includes SpectrumView only because it needs SpectralData.  They both do, so you can make separate new files just for that structure, and the cycle is broken.  It is a common lower-level dependency of both.

Better:  avoid a header including a header.  Make `class SpectralData;` just an incomplete type definition here, and include SpectralData.h only in BrushHandle.cpp.

Maybe you will try that and find a compilation failure -- because the compiler tries to generate an in-lin destructor of BrushHandle, which requires a dtor of `shared_ptr<SpectralData>`, which requires the dtor of SpectralData, which is only forward-declared.

And so you fix that by explicitly declaring the destructor and making it non-inline.  Maybe you will need the same for copy and assignments, or instead you can also fix that by `= delete;` - ing them.